### PR TITLE
Initial release of API Template/Helpers for ASPNET-Core Mvc websites

### PR DESCRIPTION
**Pull request by PureKrome**
- Readme with samples.
- All responses are JSON.
- Common error schema/model.
- Auto model binding using fluent val.

### CODE REVIEW

**Comment by PureKrome** [comment]
👀 @RPM1984

File: README.md [hunk 1]
```diff
@@ -2,24 +2,173 @@
...
+    services.AddCustomFluentValidation(types);
+}
+```
 
 ---
 
 ## Contributing
 
-Discussions and pull requests are encouraged :) Please ask all general questions in this repo or pick a specialized repo for specific, targetted issues. We also have a [contributing](https://github.com/OpenRealEstate/OpenRealEstate/blob/master/CONTRIBUTING.md) document which goes into detail about how to do this.
+Discussions and pull requests are encouraged :) Please ask all general questions in this repo or pick a specialized repo for specific, targetted issues. We also have a [contributing](https://github.com/OpenRealEstate/Homely/blob/master/CONTRIBUTING.md) document which goes into detail about how to do this.
```
Comment (RPM1984) [inline README.md]: update URL

File: tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/GetNotFoundTests.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+using System.Net;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Homely.AspNetCore.Mvc.Helpers.Tests.TestControllerTests
+{
+    public class GetNotFoundTests : TestSetup
+    {
+        [Fact]
+        public async Task GivenABadRoute_Get_ReturnsAnHttp400()
```
Comment (RPM1984) [inline tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/GetNotFoundTests.cs]: `GivenABadRoute_Get_ReturnsAnHttp404`

File: tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/GetTests.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            // Act.
+            var response = await Client.GetAsync($"/test/{id}");
+
+            // Assert.
+            response.IsSuccessStatusCode.ShouldBeTrue();
+            await response.Content.ShouldLookLike(expectedFakeVehicle);
+        }
+
+        [Fact]
+        public async Task GivenAnInvalidId_Get_ReturnsAnHttp400()
```
Comment (RPM1984) [inline tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/GetTests.cs]: `GivenAnInvalidId_Get_ReturnsAnHttp404`

File: tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/ErrorTests.cs [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            // We want to use our own startup stuff so we can define our error callback.
+            var server = new TestServer(new WebHostBuilder().UseStartup<TestStartupWithCors>());
+            var client = server.CreateClient();
+            client.DefaultRequestHeaders.Add("Origin", "http://foo.example");
+
+            // Act.
+            var response = await client.GetAsync("/test/error");
+
+            // Assert.
+            response.StatusCode.ShouldBe(HttpStatusCode.InternalServerError);
```
Comment (RPM1984) [inline tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/ErrorTests.cs:105]: should this be 500? not really a _server_ error?

**Review by RPM1984 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (PureKrome) [reply README.md]: AH! that was offscreen, which was how I missed it :(

File: tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/GetNotFoundTests.cs (hunk 2 above)
Comment (PureKrome) [reply tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/GetNotFoundTests.cs]: 🔧

File: tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/GetTests.cs (hunk 3 above)
Comment (PureKrome) [reply tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/GetTests.cs]: 🔧

File: tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/ErrorTests.cs (hunk 4 above)
Comment (PureKrome) [reply tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/ErrorTests.cs:105]: I don't understand what you are asking?

This test (`GivenAnAjaxGetRequestWithCorsEnabled_Error_ReturnsAnHttp500`) is a test to see that we can return the correct CORS headers, given the request is a CORS request _and_ an `InternalServerError` (HTTP 500 error) has occurred.

Comment (RPM1984) [reply tests/Homely.AspNetCore.Mvc.Helpers.Tests/TestControllerTests/ErrorTests.cs:105]: Ah, i see now. I got confused. I thought you were testing here than when the CORS header was _rejected_, it should return a 500. (should be a 401/403). All good 😄

**Review by RPM1984 — APPROVED** [review]

**Pull request merged**